ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/auth/OngoingStreamAuthIT.scala

```diff
@@ -0,0 +1,149 @@
+
+    def observeTransactionsStream(
+        token: Option[String],
+        party: String,
+    ): StreamObserver[GetTransactionsResponse] = {
+      val observer = new StreamObserver[GetTransactionsResponse] {
+        override def onNext(value: GetTransactionsResponse): Unit = {
+          val _ = receivedTransactionsCount.incrementAndGet()
+        }
+
+        override def onError(t: Throwable): Unit = {
+          val _ = transactionStreamAbortedPromise.trySuccess(t)
+        }
+
+        override def onCompleted(): Unit = ()
+      }
+      val request = new GetTransactionsRequest(
+        ledgerId = unwrappedLedgerId,
+        begin = Option(ledgerBegin),
+        end = None,
+        filter = Some(
+          new TransactionFilter(
+            Map(party -> new Filters)
+          )
+        ),
+      )
+      stub(TransactionServiceGrpc.stub(channel), token)
+        .getTransactions(request, observer)
+      observer
+    }
+
+    val canActAsAlice = Right(Right.Kind.CanActAs(Right.CanActAs(partyAlice)))
+    for {
+      (userAlice, tokenAlice) <- createUserByAdmin(
+        userId = userIdAlice,
+        rights = Vector(canActAsAlice),
+      )
+      applicationId = userAlice.id
+      submitAndWaitF = () =>
+        submitAndWait(token = tokenAlice, party = partyAlice, applicationId = applicationId)
+      _ <- submitAndWaitF()
+      streamObserver = observeTransactionsStream(tokenAlice, partyAlice)
+      _ <- submitAndWaitF()
+      // Making a change to the user Alice
+      _ <- grantUserRightsByAdmin(
+        userId = userIdAlice,
+        Right(Right.Kind.CanActAs(Right.CanActAs(UUID.randomUUID().toString))),
+      )
+      _ = Thread.sleep(UserManagementCacheExpiryInSeconds.toLong * 1000)
+      //
+      _ <- submitAndWaitF()
+      _ <- submitAndWaitF()
+      _ <- submitAndWaitF()
+      // Timer that finishes the stream in case it isn't aborted as expected
+      timerTask = new TimerTask {
+        override def run(): Unit = streamObserver.onError(
+          new AssertionError("Timed-out waiting while waiting for stream to abort")
+        )
+      }
+      _ = new Timer(true).schedule(timerTask, 100)
+      t <- transactionStreamAbortedPromise.future
+    } yield {
+      timerTask.cancel()
+      t match {
+        case sre: StatusRuntimeException =>
+          assertError(
+            actual = sre,
+            expectedCode = Status.Code.ABORTED,
+            expectedMessage =
+              "STALE_STREAM_CLAIMS(2,0): Authentication claims out of date. Retry quickly.",
+            expectedDetails = List(
+              ErrorDetails.ErrorInfoDetail(
+                "STALE_STREAM_CLAIMS",
+                Map(
+                  "participantId" -> "'sandbox-participant'",
+                  "category" -> "2",
+                  "definite_answer" -> "false",
+                ),
+              ),
+              ErrorDetails.RetryInfoDetail(retryDelayInSeconds = 1),
```

Comment:
Why do we have a 1 second delay here? I'd expect we want to recommend an immediate retry wouldn't we?

Reply:
Changed to `0.seconds`